### DISCUSSION TITLE
Typescript Collection Types codegen

### DISCUSSION THREAD

**Discussion by Be-P** [directus/directus#20526] [Feature Requests]
### Summary

It would be great to have type generation based on the existing collections 

### Basic Example

- create a collection 
- directus cli codegen 
- import the generated code in ts extensions

### Motivation

It is unpleasant to keep typing aligned manually

### Detailed Design

I don't honestly know what implementation detail write down here

### Must Have:
- codegen cli command

### Drawbacks

I don't think there are many drawbacks

### Alternatives

Manually write types 

### Adoption Strategy

The adoption is straight forward

**Comment by rijkvanzanten** [comment]
Duplicate of https://github.com/directus/directus/issues/18121
